### PR TITLE
Add sample request metrics to access logging service

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Added additional metrics reporting and examples to the gRPC access logging service. This service is intended to be built upon to handle specific needs for reporting and observability in different advanced use cases.

# Context

Some users wanted more advanced metrics labels/dimensions on requests, and needed some guidance and how to do so. This PR updates our example gRPC access logging service to show more explicitly some of the kinds of things that can be done in this service.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# Testing:

Ran the access logger against our documented jwt and transformation workflows (as directed in code comments) to ensure we are getting the desired information from the filter metadata. Deployed the default access logger as of these changes and ensured that the metrics were being scraped and that the times recorded were sane.

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2804

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick ingress test flake https://console.cloud.google.com/cloud-build/builds/ef5a1cc8-3d7c-4db6-8944-1633aafbd1e2;step=16?project=solo-public

**Comment by kdorosh** [comment]
/kick mix of invalid and valid vs https://console.cloud.google.com/cloud-build/builds/99f010a3-c0db-49dd-80e6-943d4f1af4cf?project=solo-public

**Comment by kdorosh** [comment]
/kick mix of valid invalid vs flake https://console.cloud.google.com/cloud-build/builds/4f1fac20-8999-4a13-a8cc-eea459b00aac?project=solo-public

**Comment by kdorosh** [comment]
/kick mix of valid invalid vs flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=de07d29f-4a99-4373-878b-ef267acc36d0

File: projects/accesslogger/pkg/runner/run.go [hunk 1]
```diff
@@ -109,3 +217,29 @@ func StartAccessLog(ctx context.Context, clientSettings Settings, service *loggi
...
+	return ""
+}
+
+func getClaimFromJwtInDynamicMetadata(claim string, filterMetadata map[string]*_struct.Struct) string {
+	providerByJwt := filterMetadata["envoy.filters.http.jwt_authn"]
+	jwts := providerByJwt.GetFields()
+	for _, jwt := range jwts {
+		claims := jwt.GetStructValue()
+		if claims != nil {
+			for c, val := range claims.GetFields() {
```
Comment (yuval-k) [inline projects/accesslogger/pkg/runner/run.go:237]: IIRC fields is a map?
so no need for iteration if that's the case

File: projects/accesslogger/pkg/runner/run.go [hunk 2]
```diff
@@ -40,18 +83,83 @@ func Run() {
...
+						downstreamRespTime := v.GetCommonProperties().GetTimeToLastDownstreamTxByte()
+						downstreamRespTimeNs := int64(downstreamRespTime.GetNanos()) + (downstreamRespTime.GetSeconds()*1 ^ 9)
+
+						timeToFirstUpstreamRxByte := v.GetCommonProperties().GetTimeToFirstUpstreamRxByte()
+						timeToFirstUpstreamRxByteNs := int64(timeToFirstUpstreamRxByte.GetNanos()) + (timeToFirstUpstreamRxByte.GetSeconds()*1 ^ 9)
+						timeToLastUpstreamTxByte := v.GetCommonProperties().GetTimeToLastUpstreamTxByte()
+						timeToLastUpstreamTxByteNs := int64(timeToLastUpstreamTxByte.GetNanos()) + (timeToLastUpstreamTxByte.GetSeconds()*1 ^ 9)
+
+						// this excludes the time filters take during the processing of the request and response
+						upstreamRespTimeNs := timeToFirstUpstreamRxByteNs - timeToLastUpstreamTxByteNs
```
Comment (yuval-k) [inline projects/accesslogger/pkg/runner/run.go:124]: comment on when this can be negative

Comment (yuval-k) [reply projects/accesslogger/pkg/runner/run.go:124]: maybe don't measure it if it is negative?

**Pull request merged**